pkg/api/store/workload/workload_store.go

```diff
@@ -410,3 +434,54 @@ func imageUpdated(apiContext *types.APIContext, data map[string]interface{}) (bo
 	}
 	return false, nil
 }
+
+func tolerationExists(workload map[string]interface{}) bool {
+	tolerations, ok := values.GetSlice(workload, "scheduling", "tolerate")
+	if !ok {
+		return false
+	}
+	for _, toleration := range tolerations {
```

Comment:
nit: I don't think you need to convert the below values before comparing them, it will just evaluate as false if they are not the same type.

Reply:
If I don't convert the values below, I will need to use `reflect.DeepEqual` to do that, which is more slow way to compare.

Reply:
```suggestion
	for _, toleration := range tolerations {
	if toleration["key"] == taints.NodeTaint.Key &&
			toleration["effect"] == string(taints.NodeTaint.Effect) &&
			toleration["value"] == taints.NodeTaint.Value &&
			toleration["operator"] == string(corev1.TolerationOpEqual) {
			return true
```